docs/evm_compatibility/overview.md

```diff
@@ -1,63 +1,155 @@
+Interoperability between Solana and Ethereum EVMs requires certain adaptations. In addition to Solana-specific restrictions, two major differences include:
 
 ### Precompiles
 Neon supports all precompiled contracts defined on [evm.code](https://www.evm.codes/precompiled?fork=merge) that provide more advanced functionalities, with [certain limitations](./precompiles#limitations) on some precompiled contracts on Neon EVM.
 
+Neon also supports native precompiled contracts that are available to our users.
+
+<!-- todo once we have the details on this can link to page -->
+
 ### Gas calculation
-The mechanism of gas consumption and calculation of gas fees on Neon EVM differ from Ethereum. Gas fees on Neon EVM are much cheaper than on Ethereum. 
+The mechanism of gas consumption and calculation of gas fees on Neon EVM differ from Ethereum. Gas fees on Neon EVM are much cheaper than on Ethereum because Solana is the settlement layer. 
 
 > Learn more about the [NEON token and how gas fees work on Neon EVM](../../docs/tokens/gas_fees.md).
 
-Several Solana-specific differences also impact smart contract development.
+### Reentrancy-safe approaches
+
+:::caution
+
+Due to the difference in how gas is calculated in Neon EVM, Solidity's transfer() and send() are NOT reentrancy safe methods.
+:::
+
+Hardcoding gas is one method that programmers may apply to prevent reentrancy attacks. Within the Ethereum/Solidity ecosystem, it is possible to default gas to 2300 gwei by using `transfer()`and `send()`. However, since gas costs are now variable, these methods are [no longer advised](https://consensys.net/diligence/blog/2019/09/stop-using-soliditys-transfer-now/).
+
+### Storage
+
+Several Solana-specific differences also impact smart contract development. In Ethereum EVMs, there are two types of accounts, both of which are associated with a storage map which can be used to read and write arbitrary data:
+
+- Basic accounts store an account balance
+- Code accounts store EVM code
+
+Solana’s Sealevel also provides two accounts which are assigned up to 256 consecutive storage slots; but they have different purposes: 
+
+ - Executable
+
+ > Executable accounts provide immutable storage of executable byte code or an (immutable) proxy address of an account which stores mutable executable byte code.
+
+ - Non-executable
+
+> Since executable accounts are immutable, their application state is stored in non-executable accounts.
+
+The differences go deeper still. In an Ethereum EVM, contracts can only read and write their own storage. In Sealevel, any account’s data can be read or written to by a contract. However, the runtime enforces that only an account’s “owner” is allowed to modify it. Changes by any other programs will be reverted and cause the transaction to fail.
+
+:::info
+
+Learn more in the [Solana wiki](https://solana.wiki/docs/).
```

Comment:
@anton-lisanin not sure who is responsible for upkeep of this wiki -- is this an authoritative resource, may I use it?

Reply:
Never seen this wiki before. I think we should not use it.